src/kraken/core/formatting/exception_traceback.py

```diff
@@ -0,0 +1,62 @@
+import logging
+import textwrap
+import traceback
+from typing import Any, Callable, List, Optional, Type, TypeVar, cast
+
+from rich import print
+
+T = TypeVar("T", bound=Callable[..., Any])
+logger = logging.getLogger(__name__)
+
+
+def exception_traceback(
+    stacktrace_depth: int = 6,
+    fix_suggestion_message: Optional[str] = None,
+    allowed_exceptions: Optional[List[Type[Exception]]] = None,
+    should_catch_exception: bool = False,
+) -> Callable[[T], T]:
+    """
+    Decorator pattern that wraps a method and handles any exceptions that are raised.
+    This method only swallows the exception if the type of the exception in the list of
+    allowed types passed into this decorator as an option field.
+    """
```

Comment:
```suggestion
    """
    Decorator pattern that wraps a method and handles any exceptions that are raised.
    This method only swallows the exception if the type of the exception is in  `allowed_exceptions`.
    """
```

we should also document what happens if no `allowed_exceptions` is provided: does it swallow _every_ or _no_ exceptions?